### PR TITLE
adding parameters to have better visibility of the build state of packages

### PR DESCRIPTION
**Pull request by sbolivarm**
Parameters for the different build phases (import, prepare, install) are being added to have better information of the build process status.

### CODE REVIEW

File: lib/autobuild/package.rb [hunk 1]
```diff
@@ -68,6 +68,10 @@ def import=(value)
         # @return [Array<EnvOp>]
         attr_reader :env
 
+        #attr_accessor :prepared
```
Comment (doudou) [inline lib/autobuild/package.rb]: Leftover ?

I'm phasing out of my style of using accessors for everything. Please use the `@ivar` form and only define the predicate accessors (that you've already defined)

File: lib/autobuild/package.rb [hunk 2]
```diff
@@ -153,16 +160,19 @@ def initialize(spec = Hash.new)
...
 
             # Define the prepare task
             task "#{name}-prepare" => "#{name}-import" do
                 isolate_errors { prepare }
+                @prepared = true
             end
             task :prepare => "#{name}-prepare"
 
             task "#{name}-build"
+
```
Comment (doudou) [inline lib/autobuild/package.rb]: Don't add random blank lines

File: lib/autobuild/package.rb [hunk 3]
```diff
@@ -185,6 +195,31 @@ def checked_out?
...
+        end
+
+        def build_invoked?
+            task("#{name}-build").already_invoked?
+        end
+
+        def built?
+            @built
+        end
+
```
Comment (doudou) [inline lib/autobuild/package.rb]: Extra blank line

File: lib/autobuild/package.rb [hunk 4]
```diff
@@ -533,6 +572,7 @@ def run(*args, &block)
             options[:env] = options.delete(:resolved_env) ||
                 (options[:env] || Hash.new).merge(resolved_env)
             Autobuild::Subprocess.run(self, *args, options, &block)
+
```
Comment (doudou) [inline lib/autobuild/package.rb]: Extra blank line

File: lib/autobuild/package.rb [hunk 5]
```diff
@@ -185,6 +190,42 @@ def checked_out?
...
 
+        def prepare_invoked?
+            task("#{name}-prepare").already_invoked?
+        end
+
+        def prepared?
+            @prepared
+        end
+
+        def prepared=(val)
```
[the remark this answers is not in the document]
Comment (sbolivarm) [reply lib/autobuild/package.rb]: I only actually use one setter, for "imported", in import.rb (the imported task here in package class doesn't seem to ever be called), I added all setters for consistency but they can be removed.

[the remark this answers is not in the document]
Comment (doudou) [reply lib/autobuild/package.rb]: You could set imported at the end of `Autobuild::Package#import` instead ? It is really mean to be done this way.

[the remark this answers is not in the document]
Comment (sbolivarm) [reply lib/autobuild/package.rb]: You are right, I missed the method. I'm updating it.

**Pull request merged**